pytorch_lightning/callbacks/progress.py

```diff
@@ -293,7 +294,9 @@ def init_test_tqdm(self) -> tqdm:
     def on_sanity_check_start(self, trainer, pl_module):
         super().on_sanity_check_start(trainer, pl_module)
         self.val_progress_bar = self.init_sanity_tqdm()
-        self.val_progress_bar.total = convert_inf(trainer.num_sanity_val_steps * len(trainer.val_dataloaders))
+        self.val_progress_bar.total = sum(
+            min(trainer.num_sanity_val_steps, len(d) if has_len(d) else float('inf')) for d in trainer.val_dataloaders
```

Comment:
this is a quite common case, can't we add a function for this like

```python
def len_or_default(to_be_checked: Any, default_length: int = int('inf')):
    if has_len(to_be_checked):
        return len(to_be_checked)
    return default_length
```

This may be an overhead now, but we really need similar things quite often

Reply:
I believe this is a repeated code here. This is already done in reset_val_dataloader. All we need is just to sum num_sanity_val_steps here once #2917 is fixed.

Reply:
agree with both of you. should we block this PR with 2917 or the other way around? Does it matter which one goes first?

Reply:
I suggest block this one. Once I get some answers there I asked, I'll fix that one tonight and then we can complete this one :)